lib/packwerk/validators/dependency_validator.rb

```diff
@@ -0,0 +1,163 @@
+          )
+        end
+      end
+
+      sig { params(package_set: PackageSet).returns(ApplicationValidator::Result) }
+      def check_acyclic_graph(package_set)
+        edges = package_set.flat_map do |package|
+          package.dependencies.map do |dependency|
+            [package.name, package_set.fetch(dependency)&.name]
+          end
+        end
+
+        dependency_graph = Graph.new(edges)
+
+        cycle_strings = build_cycle_strings(dependency_graph.cycles)
+
+        if dependency_graph.acyclic?
+          ApplicationValidator::Result.new(ok: true)
+        else
+          ApplicationValidator::Result.new(
+            ok: false,
+            error_value: <<~EOS
+              Expected the package dependency graph to be acyclic, but it contains the following circular dependencies:
+
+              #{cycle_strings.join("\n")}
+            EOS
+          )
+        end
+      end
+
+      sig { params(configuration: Configuration).returns(ApplicationValidator::Result) }
+      def check_valid_package_dependencies(configuration)
+        packages_dependencies = package_manifests_settings_for(configuration, "dependencies")
+          .delete_if { |_, deps| deps.nil? }
+
+        packages_with_invalid_dependencies =
+          packages_dependencies.each_with_object([]) do |(package, dependencies), invalid_packages|
+            invalid_dependencies = if dependencies.is_a?(Array)
+              dependencies.filter { |path| invalid_package_path?(configuration, path) }
+            else
+              []
+            end
+            invalid_packages << [package, invalid_dependencies] if invalid_dependencies.any?
+          end
+
+        if packages_with_invalid_dependencies.empty?
+          ApplicationValidator::Result.new(ok: true)
+        else
+          error_locations = packages_with_invalid_dependencies.map do |package, invalid_dependencies|
+            package ||= configuration.root_path
+            package_path = Pathname.new(package).relative_path_from(configuration.root_path)
+            all_invalid_dependencies = invalid_dependencies.map { |d| "  - #{d}" }
+
+            <<~EOS
+              \t#{package_path}:
+              \t#{all_invalid_dependencies.join("\n\t")}
+            EOS
+          end
+
+          ApplicationValidator::Result.new(
+            ok: false,
+            error_value: "These dependencies do not point to valid packages:\n\n#{error_locations.join("\n")}"
+          )
+        end
+      end
+
+      sig { params(configuration: Configuration, path: T.untyped).returns(T::Boolean) }
+      def invalid_package_path?(configuration, path)
+        # Packages at the root can be implicitly specified as "."
+        return false if path == "."
+
+        package_path = File.join(configuration.root_path, path, PackageSet::PACKAGE_CONFIG_FILENAME)
+        !File.file?(package_path)
+      end
+
+      sig { params(reference: Reference).returns(String) }
+      def standard_help_message(reference)
```

Comment:
I think this is from the checker. Can we remove this method, and then fix the redundant chomp in the checker?